plugins/azure_ad_admin/icon_azure_ad_admin/actions/list_group_members/action.py

```diff
@@ -0,0 +1,63 @@
+import insightconnect_plugin_runtime
+from .schema import ListGroupMembersInput, ListGroupMembersOutput, Input, Output, Component
+
+# Custom imports below
+import requests
+from insightconnect_plugin_runtime.exceptions import PluginException
+
+
+class ListGroupMembers(insightconnect_plugin_runtime.Action):
+    def __init__(self):
+        super(self.__class__, self).__init__(
+            name="list_group_members",
+            description=Component.DESCRIPTION,
+            input=ListGroupMembersInput(),
+            output=ListGroupMembersOutput(),
+        )
+
+    def run(self, params={}):
+        # START INPUT BINDING - DO NOT REMOVE - ANY INPUTS BELOW WILL UPDATE WITH YOUR PLUGIN SPEC AFTER REGENERATION
+        group_id = params.get(Input.GROUP_ID)
+        # END INPUT BINDING - DO NOT REMOVE
+        url = f"https://graph.microsoft.com/v1.0/groups/{group_id}/members?$count=true"
+
+        # Add ConsistencyLevel header for counting
+        headers = self.connection.get_headers(self.connection.get_auth_token())
+        headers["ConsistencyLevel"] = "eventual"
+
+        response = requests.request(
+            method="GET",
+            url=url,
+            headers=headers,
+        )
+
+        if response.status_code == 404:
+            raise PluginException(
+                cause="Group not found.",
+                assistance=f"Group ID {group_id} was not found.",
+                data=response.text,
+            )
+        if not response.status_code == 200:
+            raise PluginException(
+                cause="Unexpected response from Azure AD.",
+                assistance=f"Received an unexpected response with status code {response.status_code}.",
+                data=response.text,
+            )
+
+        try:
+            result = response.json()
+            members = result.get("value", [])
+            member_count = result.get(
+                "@odata.count", len(members)
+            )  # Fallback to len(members) if @odata.count is missing
+
+            # Replace NULL values with empty strings in member attributes
+            for member in members:
+                for key, value in member.items():
+                    if value is None:
+                        member[key] = ""
```

Comment:
Depending on what you want to achieve here, we have some helper functionality to clean lists & dicts by removing empty values - it can be imported by using 
`from insightconnect_plugin_runtime.helpers import clean_list`
```
    """
    Returns a new but cleaned list.

    * None type values are removed
    * Empty string values are removed

    This function is designed so we only return useful data
    """
```

if you would rather keep the empty values for things like schema validation then what you have is fine

Reply:
Changes Implemented